web/src/beta/lib/core/Crust/Widgets/Widget/builtin/Navigator/hooks.ts

```diff
@@ -26,16 +25,14 @@ export default function ({
   onCameraOrbit?: (orbit: number) => void;
   onCameraRotateRight?: (radian: number) => void;
   onFlyTo?: (target: string | FlyToDestination, options?: { duration?: number }) => void;
-  onVisibilityChange?: (id: string, visible: boolean) => void;
+  onVisibilityChange?: () => void;
 }) {
   const [degree, setDegree] = useState(0);
   const [isHelpOpened, setIsHelpOpened] = useState(false);
   const orbitRadianRef = useRef(0);
   const isMovingOrbit = useRef(false);
-  const visible = useVisible({
-    widgetId: widget.id,
+  useVisible({
```

Comment:
Not quite understanding why we need these useVisible functions being called inside of the widgets when we are checking, and filtering out, any invisible widgets in WAS's Area component.
Can you explain that here or on slack? 🙏🏻

Reply:
Removed. I think it's unnecessary now.